src/extension/netcdf-out/src/test/java/org/geoserver/wcs2_0/MultiDimDataStore.java

```diff
@@ -94,377 +83,304 @@
+                                prop = (PropertyName) filter.getExpression1();
+                                lit = (Literal) filter.getExpression2();
+                            } else if (filter.getExpression2() instanceof PropertyName && filter.getExpression1() instanceof Literal) {
+                                prop = (PropertyName) filter.getExpression1();
+                                lit = (Literal) filter.getExpression2();
+                            } else {
+                                return super.visit(filter, data);
+                            }
+                            if (prop.getPropertyName().equals(TIME_ATTRIBUTE)) {
+                                date.setTime(((Date) lit.getValue()).getTime());
+                            }
+                            if (prop.getPropertyName().equals(BAND_DIMENSION)) {
+                                band.add((String) lit.getValue());
+                            }
+                            return super.visit(filter, data);
+                        }
+
+                        @Override
+                        public Object visit(PropertyIsBetween filter, Object data) {
+                            PropertyName prop = (PropertyName) filter.getExpression();
+
+                            if (prop.getPropertyName().equals(TIME_ATTRIBUTE)) {
+                                beginDate.setTime(((Date) ((Literal)filter.getLowerBoundary()).getValue()).getTime());
+                                endDate.setTime(((Date) ((Literal)filter.getUpperBoundary()).getValue()).getTime());
+                            }
+                            if (prop.getPropertyName().equals(BAND_DIMENSION)) {
+                                String bands = (String) ((Literal)filter.getLowerBoundary()).getValue();
+                                String[] singleBands = bands.split(",");
+                                band.addAll(Arrays.asList(singleBands));
+                            }
+                            return super.visit(filter, data);
+                        }
+
+                    };
+                    Envelope2D bbox = new Envelope2D(DefaultGeographicCRS.WGS84,-180,-90,360,180);
+
+                    query.getFilter().accept(filterVisitor, bbox);
+                    LOGGER.fine("Mosaic query on bbox: " + bbox);
+
+                    //very rudimentary filtering, for unit test only!
+                    if(bbox.getMaxX()<=35.){
+                        return wrapAndCache(Collections.singletonList(feature1));
+                    }else{
+                        return wrapAndCache(Arrays.asList(feature1,feature2));
+                    }
+
+                } catch (Exception e) {
+
+                    e.printStackTrace();
+                    throw new RuntimeException(e);
+                }
+            }
 
+            @Override
+            protected SimpleFeatureType buildFeatureType() throws IOException {
+                return null;
```

Comment:
Whoops, missed that one when autogenerating methods. Definitely should return feature type here.